### PR TITLE
[K4.0] Feature: Topic icons improvements (use category setting) #2899

### PR DESCRIPTION
**Pull request by xillibit**

### CODE REVIEW

File: libraries/kunena/template/template.php [hunk 1]
```diff
@@ -556,11 +570,11 @@ public function getImagePath($filename='', $url = true)
 		return $this->getFile($filename, $url, $this->pathTypes['images'], 'media/kunena/images');
 	}
 
-	public function getTopicIcons($all = false, $checked = 0)
+	public function getTopicIcons($all = false, $checked = 0, $category_iconset='')
 	{
 		if (empty($this->topicIcons))
 		{
-			$xmlfile = $this->getTopicIconPath('topicicons.xml', false);
+			$xmlfile = JPATH_ROOT . '/media/kunena/topic_icons/default/topicicons.xml';
```
Comment (810) [inline libraries/kunena/template/template.php]: should be:

```` $xmlfile = JPATH_ROOT . '/media/kunena/topic_icons/'. $category_iconset .'/topicicons.xml'; ````

File: libraries/kunena/template/template.php [hunk 2]
```diff
@@ -539,16 +544,25 @@ public function getRankPath($filename='', $url = false)
 		return $this->getFile($filename, $url, $this->pathTypes['ranks'], 'media/kunena/ranks');
 	}
 
-	public function getTopicIconPath($filename='', $url = true)
+	public function getTopicIconPath($filename='', $url = true, $category_iconset= '')
 	{
-		$set = $this->isHmvc() ? '/default' : '';
-		return $this->getFile($filename, $url, $this->pathTypes['topicicons'].$set, 'media/kunena/topic_icons/default');
+		if ( !empty($category_iconset) )
```
Comment (810) [inline libraries/kunena/template/template.php]:
````
$category_iconset = '/default/';   (here must be the iconset option should be loaded)
````

**Comment by xillibit** [comment]
Try now

**Comment by 810** [comment]
could you update it again

**Comment by xillibit** [comment]
I will look at it

**Comment by xillibit** [comment]
Updated

**Comment by xillibit** [comment]
Branch updated, on new topic topicons are updated when you select different catégories

**Comment by 810** [comment]
+1

**Pull request merged**